content/posts/blog/2024-03-13-image-and-distribution-1-1.md

```diff
@@ -0,0 +1,187 @@
+        "org.opencontainers.image.description": "Another example artifact for same image"
+      },
+      "artifactType": "application/example2"
+    }
+  ]
+}
+```
+
+# Support For Existing Registries
+
+The design of the changes to the image and distribution specifications assume that it will take time for registries to upgrade.
+To facilitate this transition period, changes to the image spec added new fields which should be ignored by registries without generating errors or modifying the content of the manifest.
+Clients pushing a manifest with a `subject` field and querying for referrers have an additional responsibility to manage and use a [fallback tagged Index when the registry does not support the referrers API][distribution-referrers-fallback].
+When a registry supports the referrers API, it will respond to pushing a manifest with the `subject` with the header `OCI-Subject` set to the digest in the subject field.
+When querying for referrers, a registry will respond with a `404` or similar error code if the API is not supported (when no referrers are found, an Index with an empty manifest list would be returned).
+
+The fallback tag is the digest of the requested subject, with the `:` replaced by a `-`.
+Clients would fetch that tag similar to pulling any other tagged image.
+For example if `GET /v2/<name>/referrers/sha256:21edd7d11800e94bae9f4...` returned a 404, clients should fallback to requesting `GET /v2/<name>/manifests/sha256-21edd7d11800e94bae9f4...`.
+The content of the fallback tag on older registries is maintained by clients to be identical to the referrers response on upgraded registries.
+
+The client requirement to manage the fallback tag is subject to race conditions since the existing content of the fallback tag must first be queried to add the new entry.
+The fallback tags also clutter the tag listing with entries that are not the images end users are searching for.
+Because of this, having registries upgrade to support the referrers API is preferred.
+
+# Data Field
+
+A `data` field was added to the definition of a [descriptor][image-descriptor] in the image specification.
+This optional field contains the base64 encoded content of the blob.
+When a client supports the field and encounters it, the additional requests to get the blob may be skipped.
+Because many clients do not support this field, and base64 encoding adds overhead, it is recommended to only use this feature for small blobs.
+
+# Manifest Maximum Size
+
+When [pushing manifests][distribution-manifest-push], many clients and registries have size limits to prevent various resource exhaustion attacks.
+The distribution specification now clarifies that a registry seeing its limit exceeded should return a 413 (payload too large) response to clients.
+And the specification also recommends that registries should support up to a 4MB image manifest (containing JSON descriptors, not the image layer content).
```

Comment:
```suggestion
And the specification also recommends that registries should support up to a 4MiB image manifest (containing JSON descriptors, not the image layer content).
```

Reply:
Thanks! I always mess that up. I should know better than to go by memory.